lib/influxer/metrics/metrics.rb

```diff
@@ -162,10 +170,50 @@ def tags
       @attributes.select { |k, _| tag_names.include?(k.to_s) }
     end
 
-    attributes :timestamp
+    def time(parse = false)
+      return unless @time
+
+      return parse_timestamp(@time) if parse
+      @time
+    end
+
+    def time=(val)
+      if val.is_a?(Date) || val.is_a?(Time)
+        @time = parse_time(val)
+      else
+        @time
+      end
+    end
 
     private
 
+    def parse_timestamp(val)
+      case client.time_precision
+      when 'ms'
```

Comment:
The idea is following: if you are not using 'ms' precision (which is default), then provide value as number (and we don't transform it); if you are using 'ms', then we'll try to transform it the following way:

```ruby
def parsed_timestamp
  case timestamp
  when Number
    timestamp.to_i
  when String
    (Time.parse(timestamp).to_f * 1000).to_i
  when Date
    (timestamp.to_time.to_f  * 1000).to_i
  when Time
   (timetamp.to_f * 1000).to_i
  end
end
```

I think, we'll add support for different precisions, but not now.